library/NotificationCenter/Model/Message.php

```diff
@@ -62,6 +62,17 @@ public function send(array $arrTokens, $strLanguage = '', array $arrAttachments
             }
         }
 
+        $cpLanguage = $cpLanguage === '' ? $GLOBALS['TL_LANGUAGE'] : $cpLanguage;
+        if (($objLanguage = Language::findByMessageAndLanguageOrFallback($this, $strLanguage)) === null) {
```

Comment:
Why did you introduce this block at all?
And why are you using `$strLangauge` in L66 and $cpLanguage in L67? Should be the same?

Reply:
Shouldn't the language switch (this whole block) take place prior to calling the hook (L55)?

Reply:
This logic was copied from the Gateways (e.g. https://github.com/terminal42/contao-notification_center/blob/main/library/NotificationCenter/Gateway/Email.php#L38) - because the language of the framework should be in worst case the same as the fallback language, shouldn't it?
And the Hook could manipulate it to an empty value - so I would prefer it after the Hook!